src/api/message_builder/input_selection/automatic.rs

```diff
@@ -108,7 +112,22 @@ pub(crate) async fn get_inputs(
 
                 for output_response in address_outputs {
                     available_inputs.push(InputSigningData {
-                        output_response: output_response.clone(),
+                        output: Output::try_from(&output_response.output)?,
+                        output_metadata: OutputMetadata {
+                            message_id: MessageId::from_str(&output_response.message_id)?,
+                            transaction_id: TransactionId::from_str(&output_response.transaction_id)?,
+                            output_index: output_response.output_index,
+                            is_spent: output_response.is_spent,
+                            milestone_index_spent: output_response.milestone_index_spent,
+                            milestone_timestamp_spent: output_response.milestone_timestamp_spent,
+                            transaction_id_spent: output_response
+                                .transaction_id_spent
+                                .map(|s| TransactionId::from_str(&s))
+                                .transpose()?,
+                            milestone_index_booked: output_response.milestone_index_booked,
+                            milestone_timestamp_booked: output_response.milestone_timestamp_booked,
+                            ledger_index: output_response.ledger_index,
+                        },
```

Comment:
Shouldn't we just impl TryFrom<&OutputResponse> for OutputMetadata{}?

Reply:
Ah probably what you meant with `There is also a repetition when creating an OutputMetadata that I'll fix later.`, do you also want to do that in another PR?

Reply:
Yeah, I'll do it now.